applications/teletype/src/teletype_bindings.erl

```diff
@@ -0,0 +1,119 @@
+-module(teletype_bindings).
+-behaviour(gen_server).
+
+-export([start_link/0]).
+-export([bind/3, bind/4
+        ,flush_mod/1
+        ,notification/1
+        ]).
+
+% gen_server callbacks
+-export([init/1
+        ,handle_call/3
+        ,handle_cast/2
+        ,handle_info/2
+        ,terminate/2
+        ,code_change/3
+        ]).
+
+-include("teletype.hrl").
+
+-define(SERVER, ?MODULE).
+
+-type event() :: {api_binary(), api_binary()}.
+-type responder() :: {module(), atom()}.
+%% Set would be better suited, but lists are more convenient
+-type responders() :: [responder()].
```

Comment:
Would it be preferable to use a set? We don't want duplicates, but lists are much easier to iterate over and append to.